app/controllers/api/providers_controller.rb

```diff
@@ -85,6 +85,21 @@ def import_vm_resource(type, id = nil, data = {})
       end
     end
 
+    def options
+      if params[:ems_type].blank?
```

Comment:
I still have concerns over this param and its discoverability. At present AFAICT it is not discoverable, meaning it will have to be documented somewhere, and I'm not totally comfortable with adding complexity to OPTIONS calls. Is it needed?